lib/hysync/marc_synchronizer/marc_parsing_methods/collection.rb

```diff
@@ -0,0 +1,59 @@
+module Hysync
+  module MarcSynchronizer
+    module MarcParsingMethods
+      module Collection
+        extend ActiveSupport::Concern
+        included do
+          register_parsing_method :add_collection
+        end
+
+        def add_collection(marc_record, holdings_marc_records, mapping_ruleset)
+          terms = extract_collection_clio_ids(marc_record, mapping_ruleset).map do |clio_id|
+             {
+              'collection_term' => {
+                'clio_id' => clio_id
+              }
+            }
+          end
+          dynamic_field_data['collection'] = terms unless terms.empty?
+          add_fallback_collection(marc_record, holdings_marc_records, mapping_ruleset)
+        end
+
+        # If this marc record has a collection clio id, extract it.
+        # This collection clio id, if present, will be in 773 $w and
+        # will be a 7 or 8 digit number that's optionally prefixed
+        # with "(NNC)".  The referenced clio record must also be a
+        # collection-level record, as indicated by MARC leader byte 7
+        # having a value of'c'.
```

Comment:
Having this comment here about "MARC leader byte 7 having a value of'c'." is confusing because that's not addressed in the code below (but is addressed elsewhere in the code).